packages/react/src/date-picker/reactDayPickerStyles.ts

```diff
@@ -4,140 +4,151 @@ import { focusStyles, highContrastOutlineStyles } from '../box';
-			borderRadius: tokens.borderRadius,
-			color: boxPalette.foregroundAction,
-			'&[disabled]': {
-				color: boxPalette.foregroundText,
-				opacity: 0.3,
-				cursor: 'not-allowed',
-			},
-			'&:not([disabled]):hover': {
-				backgroundColor: boxPalette.backgroundShade,
-				color: boxPalette.foregroundText,
-				textDecoration: 'underline',
-			},
-			...focusStyles,
-		},
-		// Table
-		'.rdp-months': {
-			display: 'flex',
-			height: '23.5rem',
+			opacity: 0.3,
+			cursor: 'not-allowed',
 		},
-		'.rdp-month': {
-			margin: `0 ${mapSpacing(1)}`,
-			'&:first-of-type': { marginLeft: 0 },
-			'&:last-of-type': { marginRight: 0 },
-		},
-		'.rdp-table': {
-			margin: 0,
-			maxWidth: `calc(${cellSize} * 7)`,
-			borderCollapse: 'collapse',
-		},
-		'.rdp-tbody': {
-			border: 0,
+		'&:not([disabled]):hover': {
+			backgroundColor: boxPalette.backgroundShade,
+			color: boxPalette.foregroundText,
+			textDecoration: 'underline',
 		},
-		'.rdp-cell': {
-			width: cellSize,
-			height: cellSize,
-			padding: 0,
-			textAlign: 'center',
+		...focusStyles,
+	},
+	// Table
+	'.rdp-months': {
+		display: 'flex',
+		height: '23.5rem',
+	},
+	'.rdp-month': {
+		margin: `0 ${mapSpacing(1)}`,
+		'&:first-of-type': { marginLeft: 0 },
+		'&:last-of-type': { marginRight: 0 },
+	},
+	'.rdp-table': {
+		margin: 0,
+		maxWidth: `calc(${cellSize} * 7)`,
+		borderCollapse: 'collapse',
+	},
+	'.rdp-tbody': {
+		border: 0,
+	},
+	'.rdp-cell': {
+		width: cellSize,
+		height: cellSize,
+		padding: 0,
+		textAlign: 'center',
+	},
+	'.rdp-day_selected:not([disabled]), .rdp-day_selected:focus:not([disabled]), .rdp-day_selected:active:not([disabled]), .rdp-day_selected:hover:not([disabled]), .rdp-day_selected:hover:not([disabled])':
+		{
+			'&:not(:focus)': highContrastOutlineStyles,
+			backgroundColor: boxPalette.selected,
+			color: boxPalette.backgroundBody,
+			fontWeight: tokens.fontWeight.bold,
 		},
-		'.rdp-day_selected:not([disabled]), .rdp-day_selected:focus:not([disabled]), .rdp-day_selected:active:not([disabled]), .rdp-day_selected:hover:not([disabled]), .rdp-day_selected:hover:not([disabled])':
-			{
-				'&:not(:focus)': highContrastOutlineStyles,
-				backgroundColor: boxPalette.selected,
-				color: boxPalette.backgroundBody,
-				fontWeight: tokens.fontWeight.bold,
-			},
+} as const;
+
+export const reactDayRangePickerStyles = (dateRange?: {
+	from?: Date;
+	to?: Date;
+}) => {
+	const startStyles = { borderRadius: '50% 0 0 50%' };
+	const endStyles = { borderRadius: '0 50% 50% 0' };
+
+	function startAndEndStyles() {
+		const { from, to } = dateRange ?? {};
+		// Use a fully rounded circle for single day date-ranges
+		if (from && to) return { borderRadius: '50%' };
+		// `react-day-picker` adds the start and end classes to days when only a start OR end date has been entered
+		// This improves the styles for when a user is selecting a date range for the first time
+		if (from && !to) return startStyles;
+		if (!from && to) return endStyles;
```

Comment:
Can simplify this (removing the vars and the conditions) to something like this:

```
return {
	borderRadius: '0',
	...(from && {
		borderBottomLeftRadius: '50%',
		borderTopLeftRadius: '50%',
	}),
	...(to && {
		borderBottomRightRadius: '50%',
		borderTopRightRadius: '50%',
	}),
};
```